java/com/google/copybara/git/AddExcludedFilesToIndex.java

```diff
@@ -89,17 +96,111 @@ void add() throws RepoException, IOException {
       repo.add().force().files(current).run();
     }
 
+    console.progress("Git Destination: Adding submodules");
     for (String addBackSubmodule : addBackSubmodules) {
       repo.simpleCommand("reset", "--", "--quiet", addBackSubmodule);
       repo.add().force().files(ImmutableList.of(addBackSubmodule)).run();
     }
   }
 
+  private static final class PathTreeLeaf {
+    private final String filename;
+    private boolean included;
+
+    private PathTreeLeaf(String filename, boolean included) {
+      this.filename = filename;
+      this.included = included;
+    }
+  }
+
+  private static final class PathTree {
```

Comment:
Could you add some comments to this class?

Reply:
Done